### PR TITLE
Anusha/appeals 15349

### PR DESCRIPTION
**Pull request by anusha-palliyil**
Resolves #{github issue number}

### Description
Please explain the changes you made here.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Go to ...

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

### Database Changes
*Only for Schema Changes*

* [ ] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ ] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [ ] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Post this PR in #appeals-schema with a summary
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

### Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
* [ ] Update Fakes
* [ ] Integrations impacting functionality are tested in Caseflow UAT

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -57,7 +58,7 @@ gem "rails", "5.2.4.6"
 gem "rainbow"
 # React
 gem "react_on_rails", "11.3.0"
-gem "redis-namespace"
+gem "redis-namespace", "1.6.0"
```
Comment (anusha-palliyil) [inline Gemfile:61]: This was an update for my local computer, do not need it for pdfjs.

File: spec/support/clear_cache.rb [hunk 2]
```diff
@@ -26,5 +26,8 @@
 
 def delete_matched(namespace:)
   redis = Redis.current
+#  redis = RedisClient.current.ping
+#  redis = Redis.new(url: "redis://appeals-redis:6379/0")
+#  redis = Redis.new(url: "redis://localhost:6379/1")
```
Comment (anusha-palliyil) [inline spec/support/clear_cache.rb:31]: This was an update for my local computer, do not need it for pdfjs.

File: Gemfile [hunk 3]
```diff
@@ -45,7 +45,8 @@ gem "parallel"
 gem "paranoia", "~> 2.2"
 # PDF Tools
 gem "pdf-forms"
-gem "pdfjs_viewer-rails", git: "https://github.com/senny/pdfjs_viewer-rails.git", ref: "a4249eacbf70175db63b57e9f364d0a9a79e2b43"
+# gem "pdfjs_viewer-rails", git: "https://github.com/senny/pdfjs_viewer-rails.git", ref: "a4249eacbf70175db63b57e9f364d0a9a79e2b43"
+gem "pdfjs_viewer-rails", git: "https://github.com/department-of-veterans-affairs/pdfjs_viewer-rails-va.git", branch: "anusha/APPEALS-14211"
```
Comment (anusha-palliyil) [inline Gemfile:49]: Will need this update.

File: Gemfile.lock [hunk 4]
```diff
@@ -55,6 +55,16 @@ GIT
...
+GIT
+  remote: https://github.com/department-of-veterans-affairs/pdfjs_viewer-rails-va.git
+  revision: 3bfb123843af39f77366c8a7862aa2bc279881a0
+  branch: anusha/APPEALS-14211
+  specs:
+    pdfjs_viewer-rails (0.3.2)
+      json (> 1.8.4)
+      rails (> 4.2.0)
+      sassc-rails (>= 2.1)
+
```
Comment (anusha-palliyil) [inline Gemfile.lock:67]: Will need this update.

File: Gemfile.lock [hunk 5]
```diff
@@ -278,7 +280,7 @@ GEM
       dry-logic (~> 1.0, >= 1.0.2)
     ecma-re-validator (0.2.1)
       regexp_parser (~> 1.2)
-    erubi (1.10.0)
+    erubi (1.12.0)
```
Comment (anusha-palliyil) [inline Gemfile.lock:283]: Automatic update, updated to the most up-to-date version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 6]
```diff
@@ -225,6 +226,7 @@ GEM
       octokit (~> 4.7)
       terminal-table (~> 1)
     database_cleaner (1.7.0)
+    date (3.3.3)
```
Comment (anusha-palliyil) [inline Gemfile.lock:229]: A new gem was added after running bundle install.

File: Gemfile.lock [hunk 7]
```diff
@@ -298,7 +300,7 @@ GEM
     fasterer (0.6.0)
       colorize (~> 0.7)
       ruby_parser (>= 3.13.0)
-    ffi (1.11.1)
+    ffi (1.15.5)
```
Comment (anusha-palliyil) [inline Gemfile.lock:303]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 8]
```diff
@@ -307,8 +309,8 @@ GEM
       ffi (~> 1.0)
     git (1.7.0)
       rchardet (~> 1.8)
-    globalid (0.4.2)
-      activesupport (>= 4.2.0)
+    globalid (1.1.0)
+      activesupport (>= 5.0)
```
Comment (anusha-palliyil) [inline Gemfile.lock:313]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 9]
```diff
@@ -338,7 +340,7 @@ GEM
     httpi (2.4.4)
       rack
       socksify
-    i18n (1.8.10)
+    i18n (1.12.0)
```
Comment (anusha-palliyil) [inline Gemfile.lock:343]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 10]
```diff
@@ -356,7 +358,7 @@ GEM
       execjs (>= 1.4.0)
       multi_json (~> 1.0)
       therubyracer (~> 0.12.1)
-    json (2.3.0)
+    json (2.6.3)
```
Comment (anusha-palliyil) [inline Gemfile.lock:361]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 11]
```diff
@@ -390,12 +392,15 @@ GEM
     logstasher (2.1.5)
       activesupport (>= 5.2)
       request_store
-    loofah (2.9.1)
+    loofah (2.19.1)
```
Comment (anusha-palliyil) [inline Gemfile.lock:395]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 12]
```diff
@@ -390,12 +392,15 @@ GEM
...
       crass (~> 1.0.2)
       nokogiri (>= 1.5.9)
     lumberjack (1.0.13)
-    mail (2.7.1)
+    mail (2.8.1)
```
Comment (anusha-palliyil) [inline Gemfile.lock:399]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 13]
```diff
@@ -390,12 +392,15 @@ GEM
...
       mini_mime (>= 0.1.1)
+      net-imap
+      net-pop
+      net-smtp
```
Comment (anusha-palliyil) [inline Gemfile.lock:403]: A new gem was added after running bundle install.

File: Gemfile.lock [hunk 14]
```diff
@@ -410,9 +415,9 @@ GEM
     mimemagic (0.3.10)
       nokogiri (~> 1)
       rake
-    mini_mime (1.1.0)
-    mini_portile2 (2.7.1)
-    minitest (5.14.4)
+    mini_mime (1.1.2)
+    mini_portile2 (2.8.1)
+    minitest (5.17.0)
```
Comment (anusha-palliyil) [inline Gemfile.lock:420]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 15]
```diff
@@ -427,11 +432,20 @@ GEM
...
     nenv (0.3.0)
+    net-imap (0.3.4)
+      date
+      net-protocol
+    net-pop (0.1.2)
+      net-protocol
+    net-protocol (0.2.1)
+      timeout
+    net-smtp (0.3.3)
+      net-protocol
```
Comment (anusha-palliyil) [inline Gemfile.lock:443]: A new gem was added after running bundle install.

File: Gemfile.lock [hunk 16]
```diff
@@ -427,11 +432,20 @@ GEM
...
     newrelic_rpm (6.5.0.357)
     nio4r (2.5.8)
     no_proxy_fix (0.1.2)
-    nokogiri (1.13.1)
-      mini_portile2 (~> 2.7.0)
+    nokogiri (1.14.2)
+      mini_portile2 (~> 2.8.0)
```
Comment (anusha-palliyil) [inline Gemfile.lock:448]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 17]
```diff
@@ -465,12 +479,12 @@ GEM
     public_suffix (4.0.6)
     puma (5.5.2)
       nio4r (~> 2.0)
-    racc (1.6.0)
-    rack (2.2.3)
+    racc (1.6.2)
+    rack (2.2.6.2)
```
Comment (anusha-palliyil) [inline Gemfile.lock:483]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 18]
```diff
@@ -465,12 +479,12 @@ GEM
...
     rack-contrib (2.1.0)
       rack (~> 2.0)
-    rack-test (1.1.0)
-      rack (>= 1.0, < 3)
+    rack-test (2.0.2)
+      rack (>= 1.3)
```
Comment (anusha-palliyil) [inline Gemfile.lock:487]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 19]
```diff
@@ -492,8 +506,8 @@ GEM
       activesupport (>= 4.2)
       choice (~> 0.2.0)
       ruby-graphviz (~> 1.2)
-    rails-html-sanitizer (1.3.0)
-      loofah (~> 2.3)
+    rails-html-sanitizer (1.5.0)
+      loofah (~> 2.19, >= 2.19.1)
```
Comment (anusha-palliyil) [inline Gemfile.lock:510]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 20]
```diff
@@ -604,6 +618,14 @@ GEM
...
       sprockets-rails (>= 2.0, < 4.0)
       tilt (>= 1.1, < 3)
+    sassc (2.4.0)
+      ffi (~> 1.9)
+    sassc-rails (2.1.2)
+      railties (>= 4.0.0)
+      sassc (>= 2.0)
+      sprockets (> 3.0)
+      sprockets-rails
+      tilt
```
Comment (anusha-palliyil) [inline Gemfile.lock:628]: A new gem was added after running bundle install.

File: Gemfile.lock [hunk 21]
```diff
@@ -648,9 +670,9 @@ GEM
     sprockets (3.7.2)
       concurrent-ruby (~> 1.0)
       rack (> 1, < 3)
-    sprockets-rails (3.2.2)
-      actionpack (>= 4.0)
-      activesupport (>= 4.0)
+    sprockets-rails (3.4.2)
+      actionpack (>= 5.2)
+      activesupport (>= 5.2)
```
Comment (anusha-palliyil) [inline Gemfile.lock:675]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 22]
```diff
@@ -664,10 +686,11 @@ GEM
       ref
     thor (0.19.4)
     thread_safe (0.3.6)
-    tilt (2.0.8)
+    tilt (2.0.11)
```
Comment (anusha-palliyil) [inline Gemfile.lock:689]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 23]
```diff
@@ -664,10 +686,11 @@ GEM
...
     timecop (0.9.1)
+    timeout (0.3.2)
```
Comment (anusha-palliyil) [inline Gemfile.lock:691]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 24]
```diff
@@ -664,10 +686,11 @@ GEM
...
     tty-tree (0.3.0)
-    tzinfo (1.2.9)
+    tzinfo (1.2.11)
```
Comment (anusha-palliyil) [inline Gemfile.lock:693]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 25]
```diff
@@ -690,7 +713,7 @@ GEM
       crack (>= 0.3.2)
       hashdiff (>= 0.4.0, < 2.0.0)
     webrick (1.7.0)
-    websocket-driver (0.7.3)
+    websocket-driver (0.7.5)
```
Comment (anusha-palliyil) [inline Gemfile.lock:716]: Automatic update, updated to the latest version. Will need this update for pdfjs.

File: Gemfile.lock [hunk 26]
```diff
@@ -778,7 +801,7 @@ DEPENDENCIES
   rainbow
   rb-readline
   react_on_rails (= 11.3.0)
-  redis-namespace
+  redis-namespace (= 1.6.0)
```
Comment (anusha-palliyil) [inline Gemfile.lock:804]: Automatic update, updated to the latest version. Will need this update for pdfjs.

**Review by anusha-palliyil — COMMENTED** [review]
Added comments to all the changed that have been made

File: Gemfile.lock (hunk 26 above)
Comment (mikefinneran) [reply Gemfile.lock:804]: This may not be related to pdfjs and might have been related to the redis setup.

File: Gemfile (hunk 3 above)
Comment (mikefinneran) [reply Gemfile:49]: Lets remove the commented out version.

File: Gemfile (hunk 1 above)
Comment (mikefinneran) [reply Gemfile:61]: This can be removed from the PR

File: Gemfile.lock (hunk 10 above)
Comment (mikefinneran) [reply Gemfile.lock:361]: This change can be removed from the PR

File: spec/support/clear_cache.rb (hunk 2 above)
Comment (mikefinneran) [reply spec/support/clear_cache.rb:31]: This can be removed from the PR

File: Gemfile (hunk 3 above)
Comment (anusha-palliyil) [reply Gemfile:49]: Commented line has been removed.

File: Gemfile (hunk 1 above)
Comment (anusha-palliyil) [reply Gemfile:61]: Changed it back to original formatting.

File: Gemfile.lock (hunk 10 above)
Comment (anusha-palliyil) [reply Gemfile.lock:361]: Changed json version back to 2.3.0

File: Gemfile.lock (hunk 26 above)
Comment (anusha-palliyil) [reply Gemfile.lock:804]: Changed redis-namespace back to original.

File: spec/support/clear_cache.rb (hunk 2 above)
Comment (anusha-palliyil) [reply spec/support/clear_cache.rb:31]: Commented lines have been removed.

**Pull request closed without merging**